x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/engines_list_flyout.tsx

```diff
@@ -0,0 +1,161 @@
+
+  const columns: Array<EuiBasicTableColumn<EnterpriseSearchEngineIndex>> = [
+    {
+      field: 'name',
+      name: i18n.translate(
+        'xpack.enterpriseSearch.content.enginesList.indicesFlyout.table.name.columnTitle',
+        {
+          defaultMessage: 'Index name',
+        }
+      ),
+      render: (indexName: string) => (
+        <EuiLinkTo
+          data-test-subj="engine-index-link"
+          data-telemetry-id="entSearchContent-engines-list-viewIndex"
+          to={generateEncodedPath(SEARCH_INDEX_PATH, { indexName })}
+        >
+          {indexName}
+        </EuiLinkTo>
+      ),
+      sortable: true,
+      truncateText: true,
+      width: '40%',
+    },
+    {
+      field: 'health',
+      name: i18n.translate(
+        'xpack.enterpriseSearch.content.enginesList.indicesFlyout.table.health.columnTitle',
+        {
+          defaultMessage: 'Index health',
+        }
+      ),
+      render: (health: 'red' | 'green' | 'yellow' | 'unavailable') => (
+        <span>
+          <EuiIcon type="dot" color={healthColorsMap[health] ?? ''} />
+          &nbsp;{health ?? '-'}
+        </span>
+      ),
+      sortable: true,
+      truncateText: true,
+      width: '15%',
+    },
+    {
+      field: 'count',
+      name: i18n.translate(
+        'xpack.enterpriseSearch.content.enginesList.indicesFlyout.table.docsCount.columnTitle',
+        {
+          defaultMessage: 'Docs count',
+        }
+      ),
+      sortable: true,
+      truncateText: true,
+      width: '15%',
+    },
+    {
+      field: 'source',
+      name: i18n.translate(
+        'xpack.enterpriseSearch.content.enginesList.indicesFlyout.table.ingestionMethod.columnTitle',
+        {
+          defaultMessage: 'Ingestion method',
+        }
+      ),
+      render: (source: IngestionMethod) => (
+        <EuiText size="s">{ingestionMethodToText(source)}</EuiText>
+      ),
+      truncateText: true,
+      width: '15%',
+    },
+  ];
+
+  if (isFetchEngineFlyoutVisible) {
+    return (
+      <EuiFlyout ownFocus aria-labelledby="enginesListFlyout" onClose={closeFetchIndicesFlyout}>
+        <EuiFlyoutHeader hasBorder>
+          <EuiTitle size="m">
+            <h2 id="engineListFlyout">
+              {i18n.translate('xpack.enterpriseSearch.content.enginesList.indicesFlyout.title', {
+                defaultMessage: 'View Indices',
+              })}
+            </h2>
+          </EuiTitle>
+          <EuiSpacer size="s" />
+          <EuiText color="subdued">
+            <FormattedMessage
+              id="xpack.enterpriseSearch.content.enginesList.indicesFlyout.subTitle"
+              defaultMessage="View the indices associated with {engineName}"
+              values={{
+                engineName: fetchEngineName,
+              }}
+            />
+          </EuiText>
+        </EuiFlyoutHeader>
+
+        <EuiFlyoutBody>
+          {engineFetchError ? (
+            <EngineError error={fetchEngineApiError} />
```

Comment:
Tested the changes. The [EngineError](https://github.com/TattdCodeMonkey/kibana/blob/31eebedae040c01673b1ea4b78d616ba2a1784b7/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_error.tsx#L21) is render correctly. But couldn't recreate scenario when error is '404'